### PR TITLE
[FEAT] Adds sync check to Goblin Village Modal prior to travel.

### LINKED ISSUE #573: [Proposal] Check sync-state prior to withdrawal function
There have been numerous reports in the Discord of individuals that are missing the text about sync needing to be done prior to withdrawal. This proposal, which I am hoping to use as my first contribution, is to add a sync-state check of some variety to the button action of the withdrawal button itself in the bank modal. Additionally, I would like to modify the text within that modal to be more "in your face" as a temporary measure during the design and testing of said function.

### PR DESCRIPTION
**Pull request by newb23**
# Description

Adds sync check to Goblin Village Modal prior to travel.

Closes #573
Continues feature from #643

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Test Suites: 12 passed, 12 total
Tests:       103 passed, 103 total
Snapshots:   0 total
Time:        19.507 s
Ran all test suites.
Done in 20.41s.

Monday, May 9, 2022 21:44:53
firefox_S0HTPRhJAG
firefox_ecHNX7EPBE
https://user-images.githubusercontent.com/7065115/167539164-7296339f-1abd-4ecc-98d1-644e56d0ad90.mp4

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Pranav-yadav** [comment]
Are you taking care of, if player has already synced on chain... In that case case player might think that sync is mandatory before every time they have to go to goblin village..


This has been discussed earlier, like is it possible to show delta if any present...
Like, "You'll loose 5 Radishes, 1 Gold, etc" if go to goblin village and craft anything"

I think someone is already working on this feat you should check it.. (if not here then in, #coders-chat)

File: src/features/farming/town/components/GoblinVillageModal.tsx [hunk 1]
```diff
@@ -13,6 +22,82 @@ export const GoblinVillageModal: React.FC<{ onClose: () => void }> = ({
     window.location.href = `${window.location.pathname}#/goblins`;
   };
 
+  const { gameService } = useContext(Context);
+  const [game] = useActor(gameService);
+  const [isLoading, setIsLoading] = useState(true);
+  const [inventory, setInventory] = useState<Inventory>({});
+  const [balance, setBalance] = useState<Decimal>(new Decimal(0));
+  const [showSyncCaptcha, setShowCaptcha] = useState(false);
+  const localInventory = JSON.stringify(
```
Comment (adamhannigan) [inline src/features/farming/town/components/GoblinVillageModal.tsx:31]: I would suggest abstracting this into a function for readability.

e.g.

```function inventoryMatch(...){
   return ....
}```

**Review by adamhannigan — COMMENTED** [review]
Nice addition! Just a small bit of feedback

Do you think we can hide the warning if they are already synced as well?

**Comment by newb23** [comment]
>Just one suggestion, you are not showing the delta in warning, showing delta will be more perfect in this case...

This feature is on my radar, yes, but will require more time to implement. 

> Do you think we can hide the warning if they are already synced as well?

This is already a part of the PR, sir!

File: src/features/farming/town/components/GoblinVillageModal.tsx (hunk 1 above)
Comment (newb23) [reply src/features/farming/town/components/GoblinVillageModal.tsx:31]: The abstraction is also on my radar, and part of what I want to combine with the comment about the Delta and specific items lost in the event of a mint/chain sync.

**Comment by spencerdezartsmith** [comment]
> # Description
> Adds sync check to Goblin Village Modal prior to travel.
> 
> Closes #573 Continues feature from #643
> 
> ## Type of change
> Please delete options that are not relevant.
> 
> * [x]  New feature (non-breaking change which adds functionality)
> 
> # How Has This Been Tested?
> Test Suites: 12 passed, 12 total Tests: 103 passed, 103 total Snapshots: 0 total Time: 19.507 s Ran all test suites. Done in 20.41s.
> 
> Monday, May 9, 2022 21:44:53 vlc_QkfAN0FPj5 firefox_Rkiafxftpx https://user-images.githubusercontent.com/7065115/167539164-7296339f-1abd-4ecc-98d1-644e56d0ad90.mp4
> 
> # Checklist:
> * [x]  Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
> * [x]  Screenshot if it includes any UI changes
> * [x]  I have read the contributing guidelines and agree to the T&Cs
> * [x]  I have performed a self-review of my own code
> * [ ]  I have commented my code, particularly in hard-to-understand areas
> * [ ]  I have made corresponding changes to the documentation
> * [x]  My changes generate no new warnings
> * [ ]  I have added tests that prove my fix is effective or that my feature works
> * [x]  New and existing unit tests pass locally with my changes
> * [ ]  Any dependent changes have been merged and published in downstream modules

Can you please pull the button out of the warning box. It would be better placed as a full width button underneath the warning. Can you also make sure that there is a consistent margin above and below the sync button. `my-2` on the button should work.

**Review by spencerdezartsmith — CHANGES_REQUESTED** [review]
Can you please pull the button out of the warning box. It would be better placed as a full width button underneath the warning. Can you also make sure that there is a consistent margin above and below the sync button. my-2 on the button should work.

**Comment by newb23** [comment]
Updated as requested with `sync` button full width with `my-2`.

**Review by spencerdezartsmith — COMMENTED** [review]
Great! I have tested this and all is working as expected. Can you please change the text to regular casing rather than uppercase so that it aligns with the other buttons. 

Also it looks like you have a lot of changes here that aren't yours. Can you make sure that you have rebased correctly with `main` so that this PR is just your changes.

**Comment by newb23** [comment]
@spencerdezartsmith 

I will address both of these comments this evening. +/- 10 hours from this message.

**Comment by newb23** [comment]
@spencerdezartsmith 
Took me a *tad* longer than 10 hours, but, we are re-based, and ready to rip!

**Comment by adamhannigan** [comment]
@newb23 Thanks for the contribution ❤️ 

We are going to close this for now as we need to do a big revamp of the syncing/progress issue in the game. Adding extra steps will make this refactor harder when the time comes

**Pull request closed without merging**